crates/router/src/connector/nuvei/transformers.rs

```diff
@@ -790,20 +791,28 @@ fn get_card_info<F>(
                         .change_context(errors::ConnectorError::MissingRequiredField {
                             field_name: "mandate_type",
                         })? {
-                        payments::MandateType::SingleUse(details) => details,
-                        payments::MandateType::MultiUse(details) => {
+                        MandateDataType::SingleUse(details) => details,
+                        MandateDataType::MultiUse(details) => {
                             details.ok_or(errors::ConnectorError::MissingRequiredField {
                                 field_name: "mandate_data.mandate_type.multi_use",
                             })?
                         }
                     };
-                    let mandate_meta: NuveiMandateMeta =
-                        utils::to_connector_meta_from_secret(Some(details.get_metadata()?))?;
+                    let mandate_meta: NuveiMandateMeta = utils::to_connector_meta_from_secret(
+                        Some(details.get_metadata().ok_or_else(utils::missing_field_err(
+                            "mandate_data.mandate_type.{multi_use|single_use}.metadata",
```

Comment:
I'm not sure if we should be returning such error messages which could possibly lead to _some_ confusion.

Reply:
yeah,
what should we replace this with?

I'm not very aware of the implications of changing this...